elastic-agent-client.proto

```diff
@@ -185,6 +185,24 @@ message FQDNFeature {
     bool enabled = 1;
 }
 
+// Component-level configuration.
+message Component {
+    ComponentLimits limits = 1;
+}
+
+// Limits to configure for the currently running component.
+message ComponentLimits {
+  // Source is the original source of the limits. All values from the limits
+  // are included here even if other concrete fields are defined for this message.
+  google.protobuf.Struct source = 1;
+
+  // The maximum number of CPUs that can be used simultaneously by the component.
+  // For Go programs, it's GOMAXPROCS, see https://pkg.go.dev/runtime#GOMAXPROCS.
+  // Other languages should set their equivalent to Go's GOMAXPROCS.
```

Comment:
As I mentioned in https://github.com/elastic/ingest-dev/issues/1656#issuecomment-1673670217 this pretty much doesn't exist and I would prefer we don't enforce this as an obligation on input/component developers.

If we can find a well know example that isn't Go we can keep the generic name, to avoid confusion I would prefer to specifically name this `go_max_procs`.

Reply:
This limit is highly specific to the Go threading model where it is implementing its own scheduler. Languages without this default to the OS level thread scheduler and controlling this isn't available from the language itself.